### ISSUE TITLE
Document Attribute Macros

### ISSUE THREAD

**Issue by mikedotexe** [near/near-sdk-rs#363]
View the current Rust docs here:
https://docs.rs/near-sdk/3.1.0/near_sdk/index.html#attributes

- [ ] `ext_contract` provide description. When you click into it
- [ ] `near_bindgen` provide description and note this is very common and core to smart contract development. When you click into it
- [ ] Extra credit for documenting other parts of this module that would benefit developers and might be frequently visited.

In the right sidebar under **Epics**, click on the link for info on running Rust docs locally and good examples of documentation.

**Comment by frol** [comment]
Just to put some colors:

Currently, the doc is sparse: https://docs.rs/near-sdk/4.0.0-pre.1/near_sdk/attr.init.html

`#[init(ignore_state)]` is not mentioned at all.

Ideally, we should talk through the implications and where those are useful (e.g. state migration: https://stackoverflow.com/questions/67041874/in-nearprotocol-how-to-migrate-contract-state)

**Comment by frol** [comment]
I recalled that I read the doc before, yet I could not find it exploring the README, the reference documentation, the links from the README, NEAR docs, and even the [new book](https://github.com/near/sdk-docs).

Here it is: https://github.com/near/near-sdk-rs/blob/master/HELP.md#upgrading-a-contract

Opened another issue: #357

**Comment by austinabell** [comment]
> #[init(ignore_state)] is not mentioned at all.

yeah, I thought it was, surprised that slipped through. These types are documented on the SDK side, so I'll move this issue to the docs repo now
